zulipterminal/model.py

```diff
@@ -1156,6 +1156,9 @@ def _handle_update_message_flags_event(self, event: Event) -> None:
             elif operation == 'remove':
                 if flag_to_change in msg['flags']:
                     msg['flags'].remove(flag_to_change)
+                if(message_id in self.index['starred_msg_ids']
+                   and flag_to_change == 'starred'):
+                    self.index['starred_msg_ids'].remove(message_id)
```

Comment:
This handles the remove case, but it seems we also are missing the add case?

Reply:
Updating shortly.